docs/RFCs/0002-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,169 @@
+- `stacktrace`: 
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+- `timestamp`:
+    The time at which the crash occurred, in ISO 8601 format.
+- `uuid`:
+    A UUID v4 which uniquely identifies the crash.
+- `version_id`:
+    An integer version id.
+
+### ErrorData
+- `threads`:
+    This **optional** field contains an array of `Thread` objects.
+    In a multi-threaded program, the collector SHOULD collect the stacktraces of all active threads, and report them here.
+    A `Thread` object has the following fields:
+    - `crashed`: a boolean which tells if the thread crashed.
+    - `name`: Name of the thread (e.g. 'Thread 0').
+    - `stack`: The `StackTrace` of the thread
+    - `state`: **Optional**. Platform-specific state of the thread when its state was captured (CPU registers dump for iOS, thread state enum for Android, etc.).
+    See below for more details on how stacktraces are formatted.
+- `is_crash`:
+    Boolean true if the error was a crash, false otherwise.
+- `kind`:
+    The kind of error that occurred.
+    For a crash, a non-exhaustive set of options include "SigAbort", "SigBus", "SigSegv".
+    This field MAY be extended to include options such at "Panic", "UnhandledException" etc.
+- `message`:
+    A human readable string containing an error message associated with the stack trace.
+- `source_type`:
+    The string "crashtracking".
+- `stack`:
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+
+### Optional fields
+Any field not listed as "Required" is optional.
+In order to minimize logging overhead, producers SHOULD NOT emit anything for an optional field.
+Consumers MUST accept json with elided optional fields.
+
+- `counters`:
+    The crashtracker offers a mechanism for programs to register counters to track which operations were active at the time of the crash.
+    At present, this is only used by the profiler, but this may be extended in the future.
+- `files`:
+    The collector MAY collect useful files, such as `/proc/self/maps` or `/proc/meminfo`, and include them here.
+    Files are stored as a `Map<filename, contents>` where `contents` is an array of plain text strings, one per line.
+- `proc_info`: 
+    Currently, just tracks the PID of the crashing process.  
+    In the future, this may record additional info about the crashing process.
+- `siginfo`:
+    The name and signal number of the crashing signal (on UNIX systems)
+- `span_ids`: 
+    A vector of string identifiers, representing the active span ids at the time of program crash.
+    The collector SHOULD collect as many as it can, but MAY cap the number of spans that it tracks.
+- `trace_ids:`
+    A vector of string identifiers, representing the active span ids at the time of program crash.
+    The collector SHOULD collect as many as it can, but MAY cap the number of spans that it tracks.
```

Comment:
Could the structure be changed to put all the top-level fields into the same paragraph, either marking each field as required/optional or creating required and optional subgroups?
The current structure is a bit confusing - one reads first about required fields, then jumps into ErrorData where some fields are also marked as optional and then gets back to optional fields.